// Physics Electron
//
// Created by Samvel Khalatyan, Apr 21, 2011
// Copyright 2011, All rights reserved

import "Physics.proto";
import "Isolation.proto";

package bsm;

message Electron
{
    message Extra
    {
        optional float d0_bsp = 1;
        optional float super_cluster_eta = 2;
        optional uint32 inner_track_expected_hits = 3;
    }

    required PhysicsObject physics_object = 1;

    optional Isolation isolation = 2;
    optional Extra extra = 3;
    optional PFIsolation pf_isolation = 4;

    enum ElectronIDName 
    {
        VeryLoose = 0;
        Loose = 1;
        Medium = 2;
        Tight = 3;
        SuperTight = 4;
        HyperTight1 = 5;
        HyperTight2 = 6;
        HyperTight3 = 7;
        HyperTight4 = 8;   
    }

    message ElectronID
    {
        required ElectronIDName name = 1;
        required bool identification = 2;
        required bool isolation = 3;
        required bool conversion_rejection = 4;
        required bool impact_parameter = 5;
    }

    repeated ElectronID electronid = 5;
}
